docs/installation/react-16.md

```diff
@@ -21,6 +21,29 @@ npm i --save-dev enzyme enzyme-adapter-react-16
 
 And then you're ready to go!  In your test files you can simply `require` or `import` enzyme:
 
+Typescript:
+
+You will need to additionally install the types files for enzyme and enzyme-adapter-react-16:
+```bash
+npm i --save-dev @types/enzyme
+npm i --save-dev @types/enzyme-adapter-react-16
+```
+
+```ts
+// setup file
+import { configure } from 'enzyme';
+import * as Adapter from 'enzyme-adapter-react-16';
```

Comment:
If Typescript requires this, it's either a massive bug, or a huge UX roadbump, in TypeScript - I would expect TypeScript to loudly and clearly document this such that any TS user should automatically be aware of it.

Separately, do you *need* to install the types files? Either way, we don't maintain the type devs in DefinitelyTyped, so referencing them here seems to be committing to more support than we're offering.

Reply:
I did some additional experimentation and you are correct the types files are not necessary and can be left out.

But i can verify that the change to import * as Adapter is necessary for typescript, otherwise the following error is generated:
```bash
TypeError: enzyme_adapter_react_16_1.default is not a constructor
```

I did notice that there is a difference between how enzyme_adapter_react_16 structures exports and how Enzyme does (and enzyme's import works as expected) 
Enzyme explicitly sets the default like this:
```js
Object.defineProperty(exports, "__esModule", {
  value: true
});

exports['default'] = shallow;
```
where as the ReactSixteenAdapter does this:
```js
module.exports = ReactSixteenAdapter;
```
perhaps changing ReactSixteenAdapter to match Enzyme's export structure would correct the issue? Unfortunately that is getting outside of my knowledge base. All I know is that currently to use the ReactSixteenAdapter with typescript you have to import it as:
```ts
import * as Adapter from 'enzyme-adapter-react-16';
```

Reply:
That's because the React 16 adapter uses the standard node pattern, where the export is `module.exports` - enzyme uses a babel-transpiled ES Module pattern, where `__esModule` is `true` and the `default` property has the real export.

TypeScript should be able to import CJS modules without `import *`; if not; that's the massive UX roadbump I was referring to in my first comment.

Reply:
TypeScript doesn't require `import *` here, strictly speaking. `import x = require(` also works, as does `const x = require(`. However, the first is incompatible with an ES2015 module target, and the second has no type information.

`import *` is definitely not valid ES2015 for pulling in a constructor, but it's the best we've got until https://github.com/Microsoft/TypeScript/issues/16093 is done.